packages/hooks/src/useOAuthConnectedApps.ts

```diff
@@ -0,0 +1,21 @@
+import { useAuthorize, useQuery } from '@deriv/api';
+
+/**
+ * A custom hook for fetching the details of connected oauth apps.
+ */
+const useOAuthConnectedApps = () => {
+    const { isSuccess } = useAuthorize();
+
+    const { data, ...rest } = useQuery('oauth_apps', {
+        options: { enabled: isSuccess },
+    });
+
+    return {
+        data: data?.oauth_apps,
+        ...rest,
+        isError: Object.hasOwn(data ?? {}, 'error'),
+        isSuccess: !Object.hasOwn(data ?? {}, 'error'),
```

Comment:
is really necessary to define `isError` & `isSuccess` explicitly as they're returned from `useQuery` alongside `data`